extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java

```diff
@@ -186,20 +192,40 @@ public static Map<String, Port> combinePorts(List<KubernetesPortBuildItem> ports
                                     : buildItemPort.getPath())
                             .build();
 
-            // Special handling for ports with name "https". We look up the container port from the Quarkus configuration.
-            if ("https".equals(name) && combinedPort.getContainerPort() == null) {
-                int containerPort = ConfigProvider.getConfig()
-                        .getOptionalValue("quarkus.http.ssl-port", Integer.class)
-                        .orElse(8443);
-
-                combinedPort = new PortBuilder(combinedPort).withContainerPort(containerPort).build();
+            // Special handling for ports with mapped configuration. We look up the container port from the Quarkus configuration.
+            if (combinedPort.getContainerPort() == null) {
+                for (Map.Entry<String, PortProperty> portWithConfig : PORTS_WITH_CONFIG.entrySet()) {
+                    if (portWithConfig.getKey().equals(name)) {
+                        combinedPort = new PortBuilder(combinedPort)
+                                .withContainerPort(portWithConfig.getValue().getPortFromConfig())
+                                .build();
+                        break;
+                    }
+                }
             }
 
             allPorts.put(name, combinedPort);
         });
         return allPorts;
     }
 
+    /**
+     * Creates the configurator build items.
+     */
+    public static void printMessageAboutPortsThatCantChange(String target, List<KubernetesPortBuildItem> ports,
+            PlatformConfiguration config) {
+        Collection<Port> allPorts = combinePorts(ports, config).values();
+        for (Port port : allPorts) {
+            for (Map.Entry<String, PortProperty> portWithConfig : PORTS_WITH_CONFIG.entrySet()) {
+                if (port.getName().equals(portWithConfig.getKey())) {
+                    LOG.info(String.format("The target '%s' is mapping the container to the port '%s' which value is '%d'. "
+                            + "You won't be able to change it at runtime using the property '%s'.",
+                            target, port.getName(), port.getContainerPort(), portWithConfig.getValue().propertyName));
```

Comment:
```suggestion
                    LOG.warn(String.format("'%s' manifests are generated with container port '%s' having value '%d'. "
                            + "The above is influenced by property '%s' which may change at runtime. Such change is likely to cause problems as the application and the container configuration will get out of sync.",
                            target, port.getName(), port.getContainerPort(), portWithConfig.getValue().propertyName));
```

Reply:
I think the message is now too long. Also, using a warning might cause some panic among users that as far as I know, never complained about this.

Reply:
let's use info instead.